src/domain/spaces/spaces.repository.ts

```diff
@@ -192,4 +215,17 @@ export class SpacesRepository implements ISpacesRepository {
 
     await spaceRepository.delete(space.id);
   }
+
+  /**
+   * Determines if a user has reached the maximum number of spaces they can create.
+   * If the user is member of a space that was not invited by anyone, they are considered to have created that space.
+   */
+  private async isLimited(userId: number): Promise<boolean> {
+    const memberRepository =
+      await this.postgresDatabaseService.getRepository(Member);
+    const unInvitedMemberships = await memberRepository.find({
+      where: { user: { id: userId }, invitedBy: IsNull() },
+    });
+    return unInvitedMemberships.length >= this.maxSpaceCreationsPerUser;
+  }
```

Comment:
Instead of relying on the absence of the `invitedBy` field, it would probably be more correct to introduce a `createdBy` property into the `Space` entity. I'd include that as a further improvement in a separate PR.